lib/cppcheck.rb

```diff
@@ -182,21 +186,57 @@ def args_builder_common()
     @config[:undefines]&.each do |undefine|
       args << "-U#{undefine}"
     end
-    
-    args += @config[:arguments]
+
+    @config[:arguments]&.each do |argument|
+      args << "#{argument}"
+    end
+
+    return args
   end
   
-  def args_builder_text()
-    return [
-      "--output-file=#{@text_artifact_filepath}"
-    ]
+  def args_builder_text(args)
+    if args.nil? || args.empty?
+      args = []
+    end
+
+    args << "--output-file=#{@text_artifact_filepath}"
+    
+    if @config[:project].nil? || @config[:project].empty?
+      COLLECTION_PATHS_INCLUDE.each do |path_include|
+        args << "-I\"#{path_include}\""
+      end
+      COLLECTION_PATHS_RELEASE_TOOLCHAIN_INCLUDE.each do |path_include|
```

Comment:
Aren't they already in `PATHS_INCLUDE`?

Reply:
The COLLECTION_PATHS_RELEASE_TOOLCHAIN_INCLUDE?

Reply:
yes, why are they needed for the analysis?

Reply:
I don't think so, but I will check.

But if they aren't, the ideia is to collect toolchain includes if they are defined.

Reply:
I'm hesitant to add `COLLECTION_PATHS_RELEASE_TOOLCHAIN_INCLUDE`, they are toolchain specific so I hardly see why it is needed to include them in the analysis.